### ISSUE TITLE
EXIF metadata strings containing multiple brackets have their values altered

### ISSUE THREAD

**Issue by lovell** [libvips/libvips#2388]
The following file contains EXIF Copyright metadata with multiple opening and closing brackets in the string value.

![test-1](https://user-images.githubusercontent.com/210965/128771922-17383495-3450-4151-b078-73aa01a401f1.jpeg)

```sh
$ exif -m test-1.jpeg | grep Copyright
Copyright	Arnold Jerocki/Divergence (Arnold Jerocki/Divergence (Photographer) - [None] (Photographer) - [None] (Editor)
```

When processed via vips this metadata is erroneously altered.

```sh
$ vips copy test-1.jpeg out.jpeg
$ exif -m out.jpeg | grep Copyright
Copyright	Arnold Jerocki/Divergence (Arnold Jerocki/Divergence (Photographer) - [None] (Arnold Jerocki/Divergence (Arnold Jerocki/Divergence (Photographer) - [None] (Photographer) - [None] (Photographer) - [None] (Editor)
```

Presumably this is due to brackets being considered "special" characters and `drop_tail` is not selecting the correct one.

https://github.com/libvips/libvips/blob/1ce301812d1c44ca6421ed93d6ba1ec4b1d14433/libvips/foreign/exif.c#L811-L818

This was originally reported at https://github.com/lovell/sharp/issues/2832

**Comment by jcupitt** [comment]
Yes, this is a problem with the way libvips handles exif metadata.

I suppose we need to quote special characters to ensure we can always get the input back again.

**Comment by lovell** [comment]
Thanks for confirming John, I wasn't sure how best to fix this. Could we use an alternative separator instead of an opening bracket? (The string is UTF-8 encoded at this point so there's plenty of choice.)

**Comment by kleisauke** [comment]
I remembered a discussion some time ago about this, with a possible alternative solution: https://github.com/kleisauke/net-vips/issues/74#issuecomment-622465228.

**Comment by jcupitt** [comment]
Fix merged to master for 8.12.
